nep-17.mediawiki

```diff
@@ -0,0 +1,107 @@
+<pre>
+  NEP: 17
+  Title: Token Standard
+  Author: Erik Zhang <erik@neo.org>
+  Type: Standard
+  Status: Draft
+  Created: 2020-11-12
+  Replaces: 5
+</pre>
+
+==Abstract==
+
+This proposal outlines a token standard for the NEO blockchain that will provide systems with a generalized interaction mechanism for tokenized Smart Contracts.  This mechanic, along with the justification for each feature are defined.  A template and examples are also provided to enable the development community.
+
+==Motivation==
+
+As the NEO blockchain scales, Smart Contract deployment and invocation will become increasingly important.  Without a standard interaction method, systems will be required to maintain a unique API for each contract, regardless of their similarity to other contracts.  Tokenized contracts present themselves as a prime example of this need because their basic operating mechanism is the same.  A standard method for interacting with these tokens relieves the entire ecosystem from maintaining a definition for basic operations that are required by every Smart Contract that employs a token.
+
+==Specification==
+
+In the method definitions below, we provide both the definitions of the functions as they are defined in the contract as well as the invoke parameters.
+
+===Methods===
+
+====symbol====
+
+<pre>
+public static string symbol()
+</pre>
+
+Returns a short string symbol of the token managed in this contract. e.g. <code>"MYT"</code>. This symbol SHOULD be short (3-8 characters is recommended), with no whitespace characters or new-lines and SHOULD be limited to the uppercase latin alphabet (i.e. the 26 letters used in English).
+
+This method MUST always return the same value every time it is invoked.
+
```

Comment:
What's the difference?

Reply:
Explicitly specifying behavior for updated contracts. It's not to replace the sentence above, but to add one.

Reply:
I think the decimals can be changed after upgrade @roman-khimov. Maybe a logic can be done to migrate balances.

Reply:
It technically can be done (especially given `_deploy` powers), but doing so will cause too much confusion and errors because the old value (considered to be stable) can be cached in lots of places.

Reply:
I think  it doesn't help the standard @roman-khimov because migration can change any aspect of the contract, including token symbol, decimals, checks for origin, total balance, everything. And, in the end, if anything goes against the standard after the migration, it means that the previous contract was NEP-5 and after migration its not, simple as that.  If one manages to change token symbol, even without migration, then it has also offended the standard, and so on.

Reply:
Maybe we will not change this for now.